go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -311,6 +361,287 @@ func registerUser(userAddress string, event []UserEventType, subsIdStr string) {
+			}
+
+			//loop through every reference address
+			for _, addr := range areaCircleCheck.Subscription.Address {
+				if addr != addressToCheck {
+					continue
+				}
+				//check if address is already inside the area or not based on the subscription
+				var withinRangeParam gisClient.TargetRange
+				withinRangeParam.Latitude = areaCircleCheck.Subscription.Latitude
+				withinRangeParam.Longitude = areaCircleCheck.Subscription.Longitude
+				withinRangeParam.Radius = areaCircleCheck.Subscription.Radius
+
+				withinRangeResp, _, err := gisAppClient.GeospatialDataApi.GetWithinRangeByName(context.TODO(), addr, withinRangeParam)
+				if err != nil {
+					log.Error("Failed to communicate with gis engine: ", err)
+					return
+				}
+
+				//check if there is a change
+				var event EnteringLeavingCriteria
+				if withinRangeResp.Within {
+					if areaCircleCheck.AddrInArea[addr] {
+						//no change
+						continue
+					} else {
+						areaCircleCheck.AddrInArea[addr] = true
+						event = ENTERING_CRITERIA
+					}
+				} else {
+					if !areaCircleCheck.AddrInArea[addr] {
+						//no change
+						continue
+					} else {
+						areaCircleCheck.AddrInArea[addr] = false
+						event = LEAVING_CRITERIA
+					}
+				}
+				//no tracking this event, stop looking for this UE
+				if *areaCircleCheck.Subscription.EnteringLeavingCriteria != event {
+					continue
+				}
+				subsIdStr := strconv.Itoa(subsId)
+				var areaCircleNotif SubscriptionNotification
+
+				areaCircleNotif.EnteringLeavingCriteria = areaCircleCheck.Subscription.EnteringLeavingCriteria
+				areaCircleNotif.IsFinalNotification = false
+				areaCircleNotif.Link = areaCircleCheck.Subscription.Link
+				var terminalLocationList []TerminalLocation
+				var terminalLocation TerminalLocation
+				terminalLocation.Address = addr
+				var locationInfo LocationInfo
+				locationInfo.Latitude = nil
+				locationInfo.Latitude = append(locationInfo.Latitude, withinRangeResp.Latitude)
+				locationInfo.Longitude = nil
+				locationInfo.Longitude = append(locationInfo.Longitude, withinRangeResp.Longitude)
+				locationInfo.Shape = 2
+				seconds := time.Now().Unix()
+				var timestamp TimeStamp
+				timestamp.Seconds = int32(seconds)
+				locationInfo.Timestamp = &timestamp
+				terminalLocation.CurrentLocation = &locationInfo
+				retrievalStatus := RETRIEVED
+				terminalLocation.LocationRetrievalStatus = &retrievalStatus
+				terminalLocationList = append(terminalLocationList, terminalLocation)
+
+				areaCircleNotif.TerminalLocation = terminalLocationList
+				areaCircleNotif.CallbackData = areaCircleCheck.Subscription.ClientCorrelator
+				var inlineCircleSubscriptionNotification InlineSubscriptionNotification
+				inlineCircleSubscriptionNotification.SubscriptionNotification = &areaCircleNotif
+				sendSubscriptionNotification(areaCircleCheck.Subscription.CallbackReference.NotifyURL, inlineCircleSubscriptionNotification)
```

Comment:
Some comment, should we be blocking the entire location service while waiting for response?

Reply:
I'm okay with leaving it as is. If we see performance degradations or longer processing delays then we can address.